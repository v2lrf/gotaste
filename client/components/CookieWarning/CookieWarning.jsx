import React, { useState, useEffect } from 'react'
import Link from 'next/link'
import Cookies from 'js-cookie'

import Container from '../Container'
import Button from '../Button'

function CookieWarning() {
  const [show, setShow] = useState(true)

  useEffect(() => {
    if (!show) {
      Cookies.set('cookies_accept', true)
    }
  })

  return Cookies.get('cookies_accept') || !show ? null : (
    <div className="fixed pin-b bg-white border-t border-red-lightest w-full py-4">
      <Container>
        <div className="flex justify-between items-center">
          <div className="text-sm mr-2 flex-shrink">
            For at give dig den bedste oplevelse hos Govinu bruger vi cookies.
            <Link href="/cookie-and-privacypolicy">
              <a className="ml-1 text-red-darker">LÃ¦s mere</a>
            </Link>
          </div>
          <div className="flex-none text-sm">
            <Button
              type="button"
              kind="secondary"
              onClick={() => setShow(false)}
            >
              Helt i orden!
            </Button>
          </div>
        </div>
      </Container>
    </div>
  )
}

export default CookieWarning
