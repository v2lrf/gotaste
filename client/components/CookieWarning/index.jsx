import dynamic from 'next/dynamic'

const CookieWarning = dynamic(() => import('./CookieWarning'), {
  ssr: false
})

export default CookieWarning
