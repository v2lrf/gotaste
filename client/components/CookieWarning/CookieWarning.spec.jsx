import React from 'react'
import CookieWarning from './CookieWarning'

describe('CookieWarning', () => {
  let component
  beforeEach(() => {
    component = render(<CookieWarning />)
  })

  describe('Snapshots', () => {
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
