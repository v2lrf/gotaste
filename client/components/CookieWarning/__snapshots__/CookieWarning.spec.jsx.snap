// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieWarning Snapshots should match the snapshot 1`] = `
<div
  class="fixed pin-b bg-white border-t border-red-lightest w-full py-4"
>
  <div
    class="container mx-auto px-4"
  >
    <div
      class="flex justify-between items-center"
    >
      <div
        class="text-sm mr-2 flex-shrink"
      >
        For at give dig den bedste oplevelse hos Govinu bruger vi cookies.
        <a
          class="ml-1 text-red-darker"
          href="/cookie-and-privacypolicy"
        >
          LÃ¦s mere
        </a>
      </div>
      <div
        class="flex-none text-sm"
      >
        <button
          class="px-4 h-8 rounded shadow bg-white text-red-darkest border border-red-lighter hover:bg-red-lightest"
          type="button"
        >
          Helt i orden!
        </button>
      </div>
    </div>
  </div>
</div>
`;
